ethereum_history_api/circuits/lib/src/rlp/rlp_type.nr

```diff
@@ -0,0 +1,49 @@
+use crate::misc::fragment::Fragment;
+use crate::rlp::decode::{decode_to_rlp_header, copy_subarray, RlpHeader, LIST, STRING};
+use crate::rlp::fragment::RlpFragment;
+
+struct FragmentWithDataType<MAX_BYTES_LEN> {
+    fragment: Fragment<MAX_BYTES_LEN>,
+    data_type: u64 // STRING or LIST
+}
+
+type RlpList<MAX_BYTES_LEN, MAX_FIELDS> = BoundedVec<FragmentWithDataType<MAX_BYTES_LEN>, MAX_FIELDS>;
+
+/// Returns an RLP list look-up table.
+/// For string elements, the offset points to the payload, whereas the offset
+///   of a list element points to the RLP header of that element.
+pub fn decode_list<MAX_BYTES_LEN, MAX_FIELDS>(input: [u8; MAX_BYTES_LEN]) -> RlpList<MAX_BYTES_LEN, MAX_FIELDS> {
```

Comment:
Why does this function live in rlp_type module?
Also - why is this module called like this? If it's supposed to store only this one type - call it by the type name

Reply:
Rearranged